util/Pos/Util/Verification.hs

```diff
@@ -0,0 +1,96 @@
+{- |
+
+Machinery to support pure datatype verification.
+
+1. Datatypes can have and not have internal pure consistency. If
+datatype doesn't have any implicit rules, it's called safe. 'Integer'
+is safe, 'Either Text String' is safe too. Otherwise, it's called
+unsafe. 'Coin' is unsafe, because it has implicit upper bound on the
```

Comment:
Can we use a different term please? Calling types safe/unsafe is really strange.
Previously some constructors were called unsafe, because calling them could lead to invariants violation.
1. Term unsafe was applied to **constructors**, not to types.
2. Now it's not the case, we can't break any invariants.

We can call types complex and simple, for example. Just a random naming idea, but it's still better than safe/unsafe, imo.

Reply:
It makes sense. Let's use checked/unchecked.